audio/softsynth/fluidsynth.cpp

```diff
@@ -601,6 +604,32 @@ MusicDevices FluidSynthMusicPlugin::getDevices() const {
 	return devices;
 }
 
+bool FluidSynthMusicPlugin::checkDevice(MidiDriver::DeviceHandle, int flags, bool quiet) const {
+#ifdef FS_HAS_STREAM_SUPPORT
+	if (flags & MDT_SUPPLIED_SOUND_FONT)
+		return true;
+#endif
+
+	Common::Path sfPath = MidiDriver_FluidSynth::getSoundFontPath();
+	if (sfPath.empty())
+		return false;
+	
+#if defined(__ANDROID__) && defined(FS_HAS_STREAM_SUPPORT)
+	if (AndroidFilesystemFactory::instance().hasSAF()) {
+		Common::FSNode fsnode(sfPath);
+		if (fsnode.exists())
+			return true;
+	}
+#endif
+
+	Common::String fsPath = sfPath.toString(Common::Path::kNativeSeparator);
+
+	struct stat st;
+	int statResult = stat(fsPath.c_str(), &st);
```

Comment:
You cannot use this in a non-backend code. Does `FSNode::isDirectory()`  not work for you?